### DISCUSSION TITLE
[New Feature] Hunter-Seeker enhancements

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1388] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

The Hunter-Seeker (HS) is arguably the most frustrating super weapon, not only in terms of vanilla gameplay, but because of its very limited modability. [Developer commentary](https://www.gamespot.com/articles/tiberian-sun-exclusive/1100-2464061/) suggests it was intended to behave quite differently: it would latch onto a random enemy unit, reveal what the unit sees, all the while doubling as a remote-controlled bomb. The spy feature was later recycled as the Limpet Drone in the Firestorm expansion.

### Proposed Code (optional)

I think the best approach is to leave aside `HunterSeeker` and instead copy some of its functions to new keys. This way the original HS behavior is retained, and modders get new modding options.

New rules keys (for UnitTypes unless stated otherwise):

`HuntRandomEnemy` Should this unit select a random enemy target on the map and pursue it, as in `Hunt` mission? Defaults to no. If set, the following apply:

- `HuntWeightedTypes` List of TechnoTypes, like `DebrisTypes`. If none are defined, all technos have equal chance of being selected as targets by this unit.
- `HuntWeights` List of weights (probability adjustments) to apply to each entry in `HuntWeightedTypes`, similar to `DebrisMaximums`. Positive values increase probability of being selected as target, negative values decrease it.
- `Suicide` (weapon flag, ported from RA2/YR) Should the unit firing this weapon die in the process? Defaults to no.

Below are some additional keys in order to restore the remote detonation feature, though I'm not sure how viable. I figured this could work via a superweapon that remotely detonates every Hunter-Seeker.

- `Type=Detonator` A new type option for super weapon types. When the SW is charged and ready, and the SW sidebar icon is clicked, the SW kills every unit with the key `RemoteDetonation` set. If the unit is attached to an enemy unit via limpet logic, damages the infected unit.
- `RemoteDetonation` (UnitType key) If set, this unit is subject to the Detonator SW.

**Comment by GeneralZAPP** [comment]
I like this idea. I support this idea.

**Comment by JoyfulShush** [comment]
Would be nice to also allow certain TechnoTypes to be excluded from selection altogether. Currently based on the suggestion, all TechnoTypes are still viable for selection, and modders would have to achieve this by reducing the weights of unwanted TechnoTypes significantly (or increasing the weights of the wanted ones), which is both tedious and also still makes it possible to select the unwanted TechnoType in very rare cases.

For completion sake, it would be nice to do it the other way around as well: make a list of the only allowed TechnoTypes - all others are excluded.

By default, all TechnoTypes are eligible for the selection.
